### PR TITLE
RS migration via AAE

### PR DESCRIPTION
**Pull request by rzezeski**
Another installment of Riak Search migration code. This pull request does the following.
- Brings the migration test up to speed with the latest code.
- Factors out some common test logic related to AAE.
- Revert the default vs. non-default bucket type change. I.e. all buckets, once again, can have indexes associated with them.
- Remove some dead code and fix type spec.

Most importantly. As of this PR it is possible to migrate a Riak Search deployment to a Yokozuna one.

There is still more to do for Riak Search migration. The biggest task being a command to iterate + index all data in a bucket or list of buckets. This is useful because migration will not require waiting for AAE and it will be useful in future for schema migrations. However, that change will require some more fundamental changes in the Yokozuna indexing code path. Rather than make an even bigger PR I thought it makes sense to push this one out so that it has a chance to make it in the 0.12.0 release.

This is related to issue #20.

### CODE REVIEW

**Comment by rzezeski** [comment]
This commit conflicts with #235. That one should land first and then this branch can be updated accordingly.

File: src/yz_app.erl [hunk 1]
```diff
@@ -83,3 +85,84 @@ setup_stats() ->
...
+
+%% @private
+%%
+%% @doc Remove Riak Search pre-commit hook from all buckets when Riak
+%% Search is disabled.
+%%
+%% Previous versions of Riak had a bug in `set_bucket' which caused
+%% bucket fixups to leak into the raw ring.  If a user is upgrading
+%% from one of these versions, and enabled search on a bucket, then
+%% the Riak Searc hook will be in the raw ring.  After migrating to
```
Comment (coderoshi) [inline src/yz_app.erl]: *Search

File: src/yz_app.erl [hunk 2]
```diff
@@ -51,11 +51,13 @@ stop(_State) ->
 maybe_setup(false) ->
     ok;
 maybe_setup(true) ->
+    Ring = yz_misc:get_ring(raw),
+    RSEnabled = yz_misc:is_riak_search_enabled(),
+    strip_rs_hooks(RSEnabled, Ring),
```
Comment (coderoshi) [inline src/yz_app.erl]: This code could be extracted into an external module, to keep this yz_app code cleaner

**Comment by coderoshi** [comment]
+1, although I didn't grok what was happening in the `yz_app` fode copied from `riak_search_kv_hook`, which I feel would be better living in its own module, since it's effectively patch code that will live only as long as RS migrations.

**Pull request merged**